adaptor-rm-webapp/src/main/java/com/sample/rm/servlet/ServletListener.java

```diff
@@ -25,27 +25,36 @@
 
 import java.net.MalformedURLException;
 import java.util.NoSuchElementException;
-import java.util.logging.Level;
-import java.util.logging.Logger;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.servlet.ServletContext;
 import javax.servlet.ServletContextEvent;
 import javax.servlet.ServletContextListener;
 import javax.servlet.ServletRegistration;
-import org.eclipse.lyo.oslc4j.core.OSLC4JUtils;
+import javax.ws.rs.core.UriBuilder;
 
+import org.eclipse.lyo.oslc4j.core.OSLC4JUtils;
 import com.sample.rm.RMToolManager;
 
 // Start of user code imports
 // End of user code
 
-public class ServletListener implements ServletContextListener  {
-    private static final String DEFAULT_BASE = "http://localhost:8080";
-    private static final String PROPERTY_BASE = servletContextParameterName("baseurl");
-    private static final Logger logger = Logger.getLogger(ServletListener.class.getName());
 
-    //If you are using another servletName in your web.xml configuration file, modify this variable early in the method contextInitialized below
-    private static String servletName = "JAX-RS Servlet";
+/**
+ * During the initialization of this ServletListener, the base URI for the OSLC resources produced by this server is configured through the OSLC4J method setPublicURI().
+ * <p>
+ * A number of approaches can be used to set this base URI:
+ * <ol>
+ * <li>Set a system environment property of name "LYO_BASE".</li>
+ * <li>In the web.xml file, set a {@code context-param} property, with name "com.sample.rm.servlet.baseurl}</li>
+ * <li>Override the value of the DEFAULT_BASE constant.</li>
```

Comment:
👍 awesome docs. Small note: `DEFAULT_BASE` no more, `fallbackBase` is used instead in your code below.